src/main/java/net/sf/jabref/model/entry/BibEntry.java

```diff
@@ -330,6 +326,11 @@ public void setField(String name, String value) {
         Objects.requireNonNull(name, "field name must not be null");
         Objects.requireNonNull(value, "field value must not be null");
 
+        if(value.isEmpty()) {
```

Comment:
I do not see any value in storing empty fields. They are not written anyway. Throwing an exception complicates things as the user might set the text to the empty string in the bibentry editor.